numpy/f2py/_isocbind.py

```diff
@@ -9,48 +9,53 @@
 
 NO WARRANTY IS EXPRESSED OR IMPLIED.  USE AT YOUR OWN RISK.
 """
+# These map to keys in c2py_map, via forced casting for now, see gh-25229
 iso_c_binding_map = {
     'integer': {
         'c_int': 'int',
-        'c_short': 'short int',
-        'c_long': 'long int',
-        'c_long_long': 'long long int',
-        'c_signed_char': 'signed char',
-        'c_size_t': 'size_t',
-        'c_int8_t': 'int8_t',
-        'c_int16_t': 'int16_t',
-        'c_int32_t': 'int32_t',
-        'c_int64_t': 'int64_t',
-        'c_int_least8_t': 'int_least8_t',
-        'c_int_least16_t': 'int_least16_t',
-        'c_int_least32_t': 'int_least32_t',
-        'c_int_least64_t': 'int_least64_t',
-        'c_int_fast8_t': 'int_fast8_t',
-        'c_int_fast16_t': 'int_fast16_t',
-        'c_int_fast32_t': 'int_fast32_t',
-        'c_int_fast64_t': 'int_fast64_t',
-        'c_intmax_t': 'intmax_t',
-        'c_intptr_t': 'intptr_t',
-        'c_ptrdiff_t': 'intptr_t',
+        'c_short': 'short',  # 'short' <=> 'int' for now
+        'c_long': 'long',  # 'long' <=> 'int' for now
+        'c_long_long': 'long_long',
+        'c_signed_char': 'signed_char',
+        'c_size_t': 'unsigned',  # size_t <=> 'unsigned' for now
```

Comment:
What's going on here is that the original PR was overly optimistic. F2PY can only support maps which are defined as part of the `c2py` keys. So for now, this performs forced casting to get around that.

Eventually, the various maps in `capi_maps` will be updated to get more correct bindings #25229. However, for now this is exactly the handling done typically at the user end via the `f2cmap` files.